File: frontend/lib/types/shipment.ts
/** Зеркало backend/app/schemas/shipment.py (только чтение — отгрузка создаётся вместе с заказом). */

interface ProductBrief {
  id: string;
  name: string;
  sku: string | null;
  unit: string;
}

interface ClientBrief {
  id: string;
  name: string;
  company_name: string | null;
}

interface OrderBrief {
  id: string;
  order_number: string;
}

interface WarehouseBrief {
  id: string;
  name: string;
}

export interface ShipmentItemRead {
  id: string;
  product_id: string;
  quantity: string;
  unit_price: string;
  total_price: string;
  product?: ProductBrief | null;
}

export interface ShipmentRead {
  id: string;
  shipment_number: string;
  order_id: string;
  client_id: string;
  warehouse_id: string;
  shipment_date: string;
  total_amount: string;
  comment: string | null;
  created_at: string;
  order?: OrderBrief | null;
  client?: ClientBrief | null;
  warehouse?: WarehouseBrief | null;
  items: ShipmentItemRead[];
}

export interface ShipmentListItem {
  id: string;
  shipment_number: string;
  order_id: string;
  client_id: string;
  shipment_date: string;
  total_amount: string;
  created_at: string;
  order?: OrderBrief | null;
  client?: ClientBrief | null;
}

File: backend/app/schemas/shipment.py
"""Схемы отгрузок (только чтение — создаются и отменяются вместе с заказом).

Отгрузка — факт продажи: создаётся атомарно вместе с заказом (см.
order_service.create_order) и сразу списывает остаток. Отдельного API для
создания/подтверждения/отмены отгрузки нет — это часть заказа.
"""
from __future__ import annotations

import uuid
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class _ProductBrief(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    sku: str | None
    unit: str


class _ClientBrief(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    company_name: str | None


class _OrderBrief(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    order_number: str


class _WarehouseBrief(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str


class ShipmentItemRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    product_id: uuid.UUID
    quantity: Decimal
    unit_price: Decimal
    total_price: Decimal
    product: _ProductBrief | None = None


class ShipmentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    shipment_number: str
    order_id: uuid.UUID
    client_id: uuid.UUID
    warehouse_id: uuid.UUID
    shipment_date: date
    total_amount: Decimal
    comment: str | None
    created_at: datetime
    order: _OrderBrief | None = None
    client: _ClientBrief | None = None
    warehouse: _WarehouseBrief | None = None
    items: list[ShipmentItemRead] = []


class ShipmentListItem(BaseModel):
    """Облегчённое представление для списка (без позиций)."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    shipment_number: str
    order_id: uuid.UUID
    client_id: uuid.UUID
    shipment_date: date
    total_amount: Decimal
    created_at: datetime
    order: _OrderBrief | None = None
    client: _ClientBrief | None = None
